src/planner/partialize.c

```diff
@@ -305,21 +305,26 @@ copy_append_like_path(PlannerInfo *root, Path *path, List *new_subpaths, PathTar
 	if (IsA(path, AppendPath))
 	{
 		AppendPath *append_path = castNode(AppendPath, path);
-		append_path->path.pathtarget = pathtarget;
-		return (Path *) copy_append_path(append_path, new_subpaths);
+		AppendPath *new_append_path = copy_append_path(append_path, new_subpaths);
+		new_append_path->path.pathtarget = pathtarget;
+		return &new_append_path->path;
 	}
 	else if (IsA(path, MergeAppendPath))
 	{
 		MergeAppendPath *merge_append_path = castNode(MergeAppendPath, path);
-		merge_append_path->path.pathtarget = pathtarget;
-		return (Path *) copy_merge_append_path(root, merge_append_path, new_subpaths);
+		MergeAppendPath *new_merge_append_path =
+			copy_merge_append_path(root, merge_append_path, new_subpaths);
+		new_merge_append_path->path.pathtarget = pathtarget;
```

Comment:
ISTM, that setting the `pathtarget` inside these `copy_*` functions is more maintainable for the future.

Reply:
Changed